src/Workspaces/Core/Portable/Extensions/OperationExtensions.cs

```diff
@@ -127,5 +156,23 @@ private static bool IsInLeftOfDeconstructionAssignment(IOperation operation)
 
             return false;
         }
+
+        public static ILocalFunctionOperation GetLocalFunctionOperation(this IOperation rootOperation, IMethodSymbol localFunction)
+        {
+            Debug.Assert(localFunction.IsLocalFunction());
+            foreach (var operation in rootOperation.DescendantsAndSelf())
+            {
+                if (operation.Kind == OperationKind.LocalFunction)
+                {
+                    var localFunctionOperation = (ILocalFunctionOperation)operation;
+                    if (localFunctionOperation.Symbol == localFunction)
+                    {
+                        return localFunctionOperation;
+                    }
+                }
+            }
+
+            return null;
```

Comment:
this feels like a it's just a .FirstOrDefault linq call.

Reply:
Sure, removed.

---
In reply to: [230606635](https://github.com/dotnet/roslyn/pull/30777#discussion_r230606635) [](ancestors = 230606635)